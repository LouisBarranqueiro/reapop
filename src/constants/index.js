export const DEFAULT_STATUS = 'default';
export const INFO_STATUS = 'info';
export const SUCCESS_STATUS = 'success';
export const WARNING_STATUS = 'warning';
export const ERROR_STATUS = 'error';

export const STATUS = {
  default: DEFAULT_STATUS,
  info: INFO_STATUS,
  success: SUCCESS_STATUS,
  warning: WARNING_STATUS,
  error: ERROR_STATUS
};

export const TOP = 't';
export const TOP_CENTER = 'tc';
export const TOP_LEFT_POSITION = 'tl';
export const TOP_RIGHT_POSITION = 'tr';
export const BOTTOM = 'b';
export const BOTTOM_CENTER = 'bc';
export const BOTTOM_LEFT_POSITION = 'bl';
export const BOTTOM_RIGHT_POSITION = 'br';

export const POSITIONS = {
  top: TOP,
  topCenter: TOP_CENTER,
  topLeft: TOP_LEFT_POSITION,
  topRight: TOP_RIGHT_POSITION,
  bottom: BOTTOM,
  bottomCenter: BOTTOM_CENTER,
  bottomLeft: BOTTOM_LEFT_POSITION,
  bottomRight: BOTTOM_RIGHT_POSITION
};

// default value for notifications
export const DEFAULT_NOTIFICATION = {
  status: DEFAULT_STATUS,
  position: TOP_RIGHT_POSITION,
  dismissible: true,
  dismissAfter: 5000,
  allowHTML: false,
  closeButton: false
};
