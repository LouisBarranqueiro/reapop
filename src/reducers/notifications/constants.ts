export enum NotificationAction {
    UpsertNotification = 'reapop/upsertNotification',
    DismissNotification = 'reapop/dismissNotification',
    DismissNotifications = 'reapop/dismissNotifications',
}
