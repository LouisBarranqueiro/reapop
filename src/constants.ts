import {NewNotification, Notification, Position} from './reducers/notifications/types'

export type NotificationConfig = {
    defaultProps: Partial<Notification>
    customizeNotification: ((notification: NewNotification) => void) | null
    generateId: ((notification: NewNotification) => string) | null
}

export const CONFIG: NotificationConfig = {
    defaultProps: {},
    customizeNotification: null,
    generateId: null,
}

export const STATUSES: {
    none: 'none'
    info: 'info'
    success: 'success'
    loading: 'loading'
    warning: 'warning'
    error: 'error'
} = Object.freeze({
    none: 'none',
    info: 'info',
    success: 'success',
    loading: 'loading',
    warning: 'warning',
    error: 'error',
})

export const POSITIONS: {
    topCenter: 'top-center'
    topLeft: 'top-left'
    topRight: 'top-right'
    bottomCenter: 'bottom-center'
    bottomLeft: 'bottom-left'
    bottomRight: 'bottom-right'
} = Object.freeze({
    topCenter: 'top-center',
    topLeft: 'top-left',
    topRight: 'top-right',
    bottomCenter: 'bottom-center',
    bottomLeft: 'bottom-left',
    bottomRight: 'bottom-right',
})

const classnamePrefix = 'reapop__'

export const classnames = {
    container: (position: Position, singleContainer: boolean) => {
        const classes = [`${classnamePrefix}container`, `${classnamePrefix}container--${position}`]
        if (singleContainer) {
            classes.push(`${classnamePrefix}container--single`)
        }
        return classes
    },
    notification: (notification: Notification) => [
        `${classnamePrefix}notification`,
        `${classnamePrefix}notification--${notification.status}`,
        `${classnamePrefix}notification--buttons-${notification.buttons.length}`,
        notification.dismissible ? `${classnamePrefix}notification--dismissible` : '',
    ],
    notificationIcon: `${classnamePrefix}notification-icon`,
    notificationImageContainer: `${classnamePrefix}notification-image-container`,
    notificationImage: `${classnamePrefix}notification-image`,
    notificationDismissIcon: `${classnamePrefix}notification-dismiss-icon`,
    notificationMeta: `${classnamePrefix}notification-meta`,
    notificationTitle: `${classnamePrefix}notification-title`,
    notificationMessage: `${classnamePrefix}notification-message`,
    notificationButtons: `${classnamePrefix}notification-buttons`,
    notificationButton: `${classnamePrefix}notification-button`,
    notificationButtonText: `${classnamePrefix}notification-button-text`,
}
