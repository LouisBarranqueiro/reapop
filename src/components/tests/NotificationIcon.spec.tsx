import React from 'react'
import {render} from '@testing-library/react'

import {Notification} from '../..'
import {POSITIONS, STATUSES} from '../../constants'
import atalhoTheme from '../../themes/atalho'

import NotificationIcon from '../NotificationIcon'

describe('<NotificationIcon/>', () => {
    const notification: Notification = {
        id: '1',
        status: STATUSES.info,
        position: POSITIONS.topLeft,
        buttons: [],
    }

    it('should render icon with given theme', () => {
        const {container} = render(<NotificationIcon notification={notification} theme={atalhoTheme} />)
        expect(container.innerHTML).toMatchSnapshot()
    })

    it.each(Object.values(STATUSES))('should render %s icon', (status) => {
        const {container} = render(<NotificationIcon notification={{...notification, status}} />)
        expect(container.innerHTML).toMatchSnapshot()
    })
})
