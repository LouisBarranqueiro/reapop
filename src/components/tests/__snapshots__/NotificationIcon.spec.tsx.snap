// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NotificationIcon/> should render error icon 1`] = `"<svg viewBox=\\"0 0 16 16\\" fill=\\"currentColor\\" class=\\"reapop__notification-dismiss-icon\\"><path fill-rule=\\"evenodd\\" d=\\"M11.46.146A.5.5 0 0 0 11.107 0H4.893a.5.5 0 0 0-.353.146L.146 4.54A.5.5 0 0 0 0 4.893v6.214a.5.5 0 0 0 .146.353l4.394 4.394a.5.5 0 0 0 .353.146h6.214a.5.5 0 0 0 .353-.146l4.394-4.394a.5.5 0 0 0 .146-.353V4.893a.5.5 0 0 0-.146-.353L11.46.146zm-6.106 4.5a.5.5 0 1 0-.708.708L7.293 8l-2.647 2.646a.5.5 0 0 0 .708.708L8 8.707l2.646 2.647a.5.5 0 0 0 .708-.708L8.707 8l2.647-2.646a.5.5 0 0 0-.708-.708L8 7.293 5.354 4.646z\\"></path></svg>"`;

exports[`<NotificationIcon/> should render icon with given theme 1`] = `"<svg viewBox=\\"0 0 16 16\\" fill=\\"currentColor\\" style=\\"display: flex; width: 20px; height: 20px; box-sizing: border-box; margin: 10px 0px 10px 15px; align-self: flex-start; flex-shrink: 0; color: rgb(65, 146, 255);\\" class=\\"reapop__notification-dismiss-icon\\"><path fill-rule=\\"evenodd\\" d=\\"M8 16A8 8 0 1 0 8 0a8 8 0 0 0 0 16zm.93-9.412l-2.29.287-.082.38.45.083c.294.07.352.176.288.469l-.738 3.468c-.194.897.105 1.319.808 1.319.545 0 1.178-.252 1.465-.598l.088-.416c-.2.176-.492.246-.686.246-.275 0-.375-.193-.304-.533L8.93 6.588zM8 5.5a1 1 0 1 0 0-2 1 1 0 0 0 0 2z\\"></path></svg>"`;

exports[`<NotificationIcon/> should render info icon 1`] = `"<svg viewBox=\\"0 0 16 16\\" fill=\\"currentColor\\" class=\\"reapop__notification-dismiss-icon\\"><path fill-rule=\\"evenodd\\" d=\\"M8 16A8 8 0 1 0 8 0a8 8 0 0 0 0 16zm.93-9.412l-2.29.287-.082.38.45.083c.294.07.352.176.288.469l-.738 3.468c-.194.897.105 1.319.808 1.319.545 0 1.178-.252 1.465-.598l.088-.416c-.2.176-.492.246-.686.246-.275 0-.375-.193-.304-.533L8.93 6.588zM8 5.5a1 1 0 1 0 0-2 1 1 0 0 0 0 2z\\"></path></svg>"`;

exports[`<NotificationIcon/> should render loading icon 1`] = `"<svg viewBox=\\"5 5 40 40\\" fill=\\"currentColor\\" class=\\"reapop__notification-dismiss-icon\\"><path d=\\"M43.935,25.145c0-10.318-8.364-18.683-18.683-18.683c-10.318,0-18.683,8.365-18.683,18.683h4.068c0-8.071,6.543-14.615,14.615-14.615c8.072,0,14.615,6.543,14.615,14.615H43.935z\\"><animateTransform attributeType=\\"xml\\" attributeName=\\"transform\\" type=\\"rotate\\" from=\\"0 25 25\\" to=\\"360 25 25\\" dur=\\"0.5s\\" repeatCount=\\"indefinite\\"></animateTransform></path></svg>"`;

exports[`<NotificationIcon/> should render none icon 1`] = `""`;

exports[`<NotificationIcon/> should render success icon 1`] = `"<svg viewBox=\\"0 0 16 16\\" fill=\\"currentColor\\" class=\\"reapop__notification-dismiss-icon\\"><path fill-rule=\\"evenodd\\" d=\\"M16 8A8 8 0 1 1 0 8a8 8 0 0 1 16 0zm-3.97-3.03a.75.75 0 0 0-1.08.022L7.477 9.417 5.384 7.323a.75.75 0 0 0-1.06 1.06L6.97 11.03a.75.75 0 0 0 1.079-.02l3.992-4.99a.75.75 0 0 0-.01-1.05z\\"></path></svg>"`;

exports[`<NotificationIcon/> should render warning icon 1`] = `"<svg viewBox=\\"0 0 16 16\\" fill=\\"currentColor\\" class=\\"reapop__notification-dismiss-icon\\"><path fill-rule=\\"evenodd\\" d=\\"M8.982 1.566a1.13 1.13 0 0 0-1.96 0L.165 13.233c-.457.778.091 1.767.98 1.767h13.713c.889 0 1.438-.99.98-1.767L8.982 1.566zM8 5a.905.905 0 0 0-.9.995l.35 3.507a.552.552 0 0 0 1.1 0l.35-3.507A.905.905 0 0 0 8 5zm.002 6a1 1 0 1 0 0 2 1 1 0 0 0 0-2z\\"></path></svg>"`;
